src/window.rs

```diff
@@ -322,10 +324,23 @@ impl WindowBuilder {
     /// See [`Window::set_always_on_top`] for details.
     #[inline]
     pub fn with_always_on_top(mut self, always_on_top: bool) -> Self {
+        self.window.always_on_bottom = false;
         self.window.always_on_top = always_on_top;
```

Comment:
Could you we refactor this to use a single `with` method and use `enum`?

```rust
pub fn with_window_layer_preference(mut self, layer: LayerPreference);

pub enum LayerPreference {
    AlwaysOnTop,
    Normal,
    AlwaysOnBottom,
}
```

Could name it differently I don't like the **layer** in the name, but you should get the idea.

Reply:
maybe:
```rs
pub fn with_window_level(mut self, level: WindowLevel) -> Self;

pub enum WindowLevel { 
  AlwaysOnTop,
  Normal,
  AlwaysOnBottom
}
```

Reply:
Yeah, that's better.